src/main/java/com/wavefront/sdk/common/application/ApplicationTags.java

```diff
@@ -82,7 +85,45 @@ public Builder shard(String shard) {
      * @return {@code this}
      */
     public Builder customTags(Map<String, String> customTags) {
-      this.customTags = customTags;
+      this.customTags.putAll(customTags);
+      return this;
+    }
+
+    /**
+     * Set additional custom tags from environment variables that matches the a
+     * regular expressions.
+     * This setting is optional.
+     *
+     * @param regex Regular expression.
+     * @return {@code this}
+     */
+    public Builder tagsFromEnv(String regex) {
+      Pattern pattern = Pattern.compile(regex, Pattern.CASE_INSENSITIVE);
+      Map<String, String> env = System.getenv();
+      env.forEach((var, value) -> {
+        Matcher matcher = pattern.matcher(var);
+        if (matcher.matches()) {
+            if (!Strings.isNullOrEmpty(value)) {
+              this.customTags.put(var, value);
+            }
+        }
+      });
+      return this;
+    }
+    
+    /**
+     * Set additional a custom tag from environment variable.
```

Comment:
`Set a custom tag from the given environment variable`. Ditto: remove the optional comment.